### PR TITLE
#364 Move ProfileUpdateBottomSheet to NavHost

### LINKED ISSUE #364: Move bottomSheet from Profile to NavHost
https://google.github.io/accompanist/navigation-material/

### PR DESCRIPTION
**Pull request by MarekMacko**
Closes #364

### CODE REVIEW

File: app/src/main/kotlin/com/intive/picover/profile/view/ProfileUpdateBottomSheet.kt [hunk 1]
```diff
@@ -16,39 +16,30 @@ import androidx.compose.material3.ModalBottomSheet
...
 @Composable
 fun ProfileUpdateBottomSheet(
 	username: String,
 	onSaveClick: () -> Unit,
-	updateVisibility: (Boolean) -> Unit,
+	onClose: () -> Unit,
 	onUsernameChange: (String) -> Unit,
 	usernameErrorMessageId: Int?,
 ) {
-	val keyboardController = LocalSoftwareKeyboardController.current
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/profile/view/ProfileUpdateBottomSheet.kt:36]: It's not needed

File: app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt [hunk 2]
```diff
@@ -64,9 +60,7 @@ fun ProfileScreen(
 				UserInfo(
 					profile = loadingProfile,
 					onEditPhotoClick = takePictureOrPickImageLauncher::launch,
-					onEditNameClick = {
-						showProfileUpdateBottomSheet = true
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt:68]: Note value of `editButtonsEnabled`, so `onEditNameClick` was not called anyway

File: app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt [hunk 3]
```diff
@@ -70,9 +62,49 @@ fun PicoverNavHost(
...
+
+@OptIn(ExperimentalMaterialNavigationApi::class)
+private fun NavGraphBuilder.profileGraph(navController: NavHostController) {
+	navigation(startDestination = "profile", route = "profileGraph") {
+		composable("profile") {
+			val parentEntry = it.rememberParentEntry(navController)
+			ProfileScreen(hiltViewModel(parentEntry), navController)
+		}
+		dialog("deleteAccount") {
+			val viewModel: ProfileViewModel = hiltViewModel()
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt:81]: `deleteAccount` should have separate view model. Anyone willing to extract related logic from `ProfileViewModel`?

Comment (BorysChajdas) [reply app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt:81]: I can handle it

**Review by BorysChajdas — APPROVED** [review]

File: app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt (hunk 3 above)
Comment (BorysChajdas) [reply app/src/main/kotlin/com/intive/picover/main/navigation/view/PicoverNavHost.kt:81]: https://github.com/Android-Guild/Picover/issues/397

**Pull request merged**